### DISCUSSION TITLE
import: right click menus: what you see is not what you get sometimes

### DISCUSSION THREAD

**Discussion by jidanni** [ImageMagick/ImageMagick#4153] [Development]
In https://github.com/openstreetmap/openstreetmap-website/issues/3309#issue-986981303 I drew a bounding box around that right click menu, and thought I
took a snapshot of it, but when I look at the snapshot, the menu is
gone! (Even though it is still sitting on the screen after the snapshot is finished.)
It turns out I only took a snapshot of the underlying map.
That was when browsing with chrome. But repeating the experiment with
firefox doesn't have that problem. Nor I suppose with chrome on my other
computer. So what odd cases produce this "what you see is not what you
get" phenomenon?  ImageMagick 6.9.11-60 (Debian sid)
Hence I need `import -window root` to finally make the image I posted.
Yes, I make the imports using the `-pause` option.

**Comment by fmw42** [comment]
What does this have to do with ImageMagick?

**Reply by jidanni** [reply]
> What does this have to do with ImageMagick?

Correct. I would give $... to figure out what it does have to do with.
Once I know what is causing the problem, we would know
why drawing a box around something sometimes doesn't capture it!!

**Reply by fmw42** [reply]
Ask on a Chrome forum. Seems to be a Chrome issue.

**Comment by jidanni** [comment]
It couldn't be a chrome issue, because I get different behavior on different computers even with the same ChromeVariations.
Same with icewm, same configuration.
So maybe it is some difference in how X windows is sending events...

**Comment by jidanni** [comment]
I mean in what cases "can I take a picture of grandma, but when I develop the film, she's gone. There are just trees there."?

**Reply by fmw42** [reply]
How graphics apps write to the image and whether that is stored in what screen grab can access.

**Reply by fmw42** [reply]
Perhaps the graphics processor is different.

**Comment by jidanni** [comment]
https://superuser.com/questions/700044/how-to-fix-disappearing-context-menus-in-google-chrome didn't help.

**Reply by fmw42** [reply]
Perhaps it depends upon the version of Chrome  and you graphics processor.

**Comment by jidanni** [comment]
OK, I am getting closer to closing in on the cause of the problem.
In https://github.com/ice-wm/icewm/issues/62 I find switching
from Normal to OnTop layer in icewm allows me to
"what I see is what I get" when taking a screenshot with Imagemagick's import function.
Now all I need is for someone to verify that is in fact a fact...
